packages/cli/src/credentials/credentials.service.ts

```diff
@@ -36,4 +96,165 @@ export class CredentialsService {
 		}
 		return new Credentials({ id: id.toString(), name }, type, nodesAccess, data);
 	}
+
+	static async prepareCredentialsCreateData(
+		data: CredentialRequest.CredentialProperties,
+	): Promise<CredentialsEntity> {
+		// Make a copy so we can delete the provided ID
+		const dataCopy = clone(data);
+		delete dataCopy.id;
```

Comment:
Or `const { id, ...rest } = data;` and drop the `id`.

Reply:
I've updated the comment here to explain why I do a deep copy here instead of a spread. It's to avoid side effects from modifying the `nodesAccess`s slightly further down the function.

Reply:
I've changed this after some changes made from Ben's review. It's now doing assignment spread.